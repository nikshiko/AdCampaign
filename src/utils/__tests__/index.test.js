import moment from 'moment'
import { generateColumnConfig, isDateInRange, isSubStringPresent } from '../index'
import columnConfig, {dateFormat} from '../../constants'
describe('utils', () => {
	test('generateColumnConfig should return the correct config',() => {
		const generatedConfig = generateColumnConfig(columnConfig)
		expect(generatedConfig).toMatchSnapshot()
	})

	test('inDateRange should return true if a given date is in the specified range',() => {
        const isInRange = isDateInRange('06-23-2019','06/22/2019','06/30/2020')
        expect(isInRange).toBeTruthy()

	})

	test('inDateRange should return false if a given date is in the specified range',() => {
		const isInRange = isDateInRange(moment().format(dateFormat),'06/21/2019','06/21/2019')
		expect(isInRange).toBeFalsy()
	})

	test('inDateRange should return false if the selected date is in the invalid format',() => {
		const isInRange = isDateInRange('12','06/21/2019','06/23/2019')
		expect(isInRange).toBeFalsy()
	})

	test('isSubStringPresent should return truthy in case the substring is present in the main string',() => {
		expect(isSubStringPresent('Sample','sa')).toBeTruthy()
		expect(isSubStringPresent('sample','z')).toBeFalsy()
	})
})