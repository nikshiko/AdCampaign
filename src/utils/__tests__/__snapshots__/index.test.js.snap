// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`utils generateColumnConfig should return the correct config 1`] = `
Array [
  Object {
    "dataIndex": "name",
    "key": "name",
    "title": "Campaign name",
  },
  Object {
    "dataIndex": "startDate",
    "key": "startDate",
    "title": "Starting on",
  },
  Object {
    "dataIndex": "endDate",
    "key": "endDate",
    "title": "Ending on",
  },
  Object {
    "dataIndex": "active",
    "key": "active",
    "title": "Active",
  },
  Object {
    "dataIndex": "Budget",
    "key": "Budget",
    "title": "Budget",
  },
]
`;
